src/ctx/locales.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Context, Result};
+use itertools::Itertools;
+
+use crate::{JError, Word};
+
+#[derive(Debug)]
+pub struct Eval {
+    pub locales: Locales,
+}
+
+#[derive(Debug)]
+pub struct Locales {
+    inner: HashMap<String, Names>,
+    search_path: Vec<String>,
+    pub anon: Vec<Names>,
+}
+
+#[derive(Debug, Default)]
+pub struct Names(pub HashMap<String, Word>);
+
+impl Eval {
+    pub fn new() -> Eval {
+        Eval {
+            locales: Locales::new(),
+        }
+    }
+}
+
+impl Locales {
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::with_capacity(8),
+            search_path: vec!["z".to_string(), "base".to_string()],
+            anon: Vec::new(),
+        }
+    }
+
+    pub fn assign_global(&mut self, n: impl ToString, v: Word) -> Result<()> {
+        let n = n.to_string();
+        let (n, ns) = parse_name(&n)?;
+        let ns = ns.unwrap_or(
+            self.search_path
+                .last()
+                .expect("non-empty search path")
+                .as_str(),
+        );
+
+        self.inner
+            .entry(ns.to_string())
+            .or_insert_with(|| Default::default())
+            .0
+            .insert(n.to_string(), v);
+        Ok(())
+    }
+
+    pub fn assign_local(&mut self, n: impl ToString, v: Word) -> Result<()> {
+        let n = n.to_string();
+        if n.contains('_') {
+            return Err(JError::NonceError).context("namespaced names");
+        }
+
+        self.anon
+            .last_mut()
+            .ok_or(JError::DomainError)
+            .context("local assignment with no local scope")?
```

Comment:
This actually shouldn't be an error.

"Local" scope also means the current locale if the current point is not within a verb:

```j
$ ijconsole
   a=.42
   a
42
   a_base_
42

```

Reply:
Locales are also how j does object oriented programming, I think I'm happy to just kick that can down the road unless you want to dig into it now?

- https://www.jsoftware.com/help/jforc/modular_code.htm#_Toc191734489
- https://code.jsoftware.com/wiki/Vocabulary/Locales#Working_with_classes_and_instances

Reply:
I fixed `=.` at the global scope by adding a new anonymous namespace at the top by default, no idea if this is a good way to model it.

Reply:
It'll do for now.  By default the repl lands in the `base` namespace in j:
- https://code.jsoftware.com/wiki/Vocabulary/Locales#The_base_locale_and_the_z-locale

But let's merge this and fix that later.